contracts/SettlementExtension.sol

```diff
@@ -77,34 +77,63 @@ contract SettlementExtension is IPostInteraction, IAmountGetter, FeeBankCharger
     ///     (bytes3,bytes2)[N] pointsAndTimeDeltas;
     /// }
 
-    function _getRateBump(bytes calldata auctionDetails) private view returns (uint256) {
-        unchecked {
-            uint256 auctionStartTime = uint32(bytes4(auctionDetails[0:4]));
-            uint256 auctionFinishTime = auctionStartTime + uint24(bytes3(auctionDetails[4:7]));
-            uint256 initialRateBump = uint24(bytes3(auctionDetails[7:10]));
-
-            if (block.timestamp <= auctionStartTime) {
-                return initialRateBump;
-            } else if (block.timestamp >= auctionFinishTime) {
-                return 0; // Means 0% bump
+    function _getRateBump(bytes calldata auctionDetails) private view returns (uint256 result) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly ("memory-safe") {
+            function timeWeightedAvg(t1, v1, t2, v2) -> avg {
+                avg := div(add(mul(sub(timestamp(), t1), v2), mul(sub(t2, timestamp()), v1)), sub(t2, t1))
```

Comment:
```suggestion
                avg := div(add(mul(sub(timestamp(), t1), v2), mul(sub(t2, timestamp()), v1)), sub(t2, t1)) // ((block.timestamp - t1) * v2 + (t2 - block.timestamp) * v1) / (t2 - t1)
```

Reply:
It is not used in fact. I forgot to remove the `timeWeightedAvg` function 😩